File: routers/update.go
package routers

import (
	"sublink/api"

	"github.com/gin-gonic/gin"
)

func Update(r *gin.Engine, version string) {
	r.GET("/api/v1/update/check", api.CheckUpdate(version))
	r.GET("/api/v1/update/status", api.UpdateStatus())
	r.POST("/api/v1/update/apply", api.ApplyUpdate())
}

File: webs/src/api/system/update.ts
import request from "@/utils/request";

export interface UpdateInfo {
  currentVersion: string;
  latestVersion: string;
  hasUpdate: boolean;
  releaseUrl: string;
  dockerImage: string;
  currentImageDigest: string;
  latestImageDigest: string;
  updateCommand: string;
  autoUpdate: boolean;
  autoUpdateMessage: string;
  containerName: string;
  message: string;
}

export interface UpdateStatus {
  status: "idle" | "running" | "success" | "rolled_back" | "failed";
  message: string;
  error?: string;
  startedAt?: string;
  finishedAt?: string;
  targetImage?: string;
  previousImage?: string;
}

export function checkUpdate() {
  return request({
    url: "/api/v1/update/check",
    method: "get",
  });
}

export function getUpdateStatus() {
  return request({
    url: "/api/v1/update/status",
    method: "get",
  });
}

export function applyUpdate() {
  return request({
    url: "/api/v1/update/apply",
    method: "post",
  });
}
